### PR TITLE
disable `pre-commit.ci` autofixing pull request branches

### PR DESCRIPTION
**Pull request by zacharyburnett**
applied suggestions from https://github.com/asdf-format/asdf/issues/1679#issuecomment-1810737318

### CODE REVIEW

File: .github/workflows/changelog.yml
Comment (braingram) [inline .github/workflows/changelog.yml:4]: This unnecessarily [exposes secrets to these jobs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) and means that PRs updating these workflows cannot be evaluated until after they are merged making managing these files more difficult.

**Review by braingram — CHANGES_REQUESTED** [review]
I am against re-enabling pre-commit-ci unless they fix the issue. The benefits it provides (autofixing prs and automatically updating pre-commit versions) do not seem worth the risk.

File: .github/workflows/changelog.yml
Comment (zacharyburnett) [reply .github/workflows/changelog.yml:4]: that's fair, it makes workflow development difficult

**Comment by WilliamJamieson** [comment]
> I am against re-enabling pre-commit-ci unless they fix the issue. The benefits it provides (autofixing prs and automatically updating pre-commit versions) do not seem worth the risk.

I fail to see how this is a risk? ASDF uses GitHub's freely available runners and all the CI that can be triggered from a PR has read only permissions (or should have read-only permissions).

**Comment by CagtayFabry** [comment]
> I am against re-enabling pre-commit-ci unless they fix the issue. The benefits it provides (autofixing prs and automatically updating pre-commit versions) do not seem worth the risk.

I am not sure if this it what you mean, but you can have at least one of the benefits (automatically updating pre-commit versions) without enabling the other (autofixing PRs) by changing the [pre-commit configuration ](https://pre-commit.ci/#configuration)in `.pre-commit-config.yaml`, e.g. by adding

```yaml
ci:
  autofix_prs: false
  autoupdate_schedule: weekly
```

**Comment by braingram** [comment]
Thanks for the suggestion. Unfortunately even with `autofix_prs` disabled adding a `pre-commit.ci autofix` comment will trigger pre-commit-ci to fix the PR which will then allow the CI to run (this comment can be added by any account, even one brand new to GitHub). It would be nice if there were a way to disable autofixing or limit permissions for auto fixes to existing maintainers but the pre-commit-ci developers aren't interested in making that change.

File: .github/workflows/changelog.yml
Comment (braingram) [reply .github/workflows/changelog.yml:4]: Reading into this a bit more the use of `pull_request_target` also provides a read/write `GITHUB_TOKEN` to the job. The [documentation](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) warns that jobs using `pull_request_target` should not check out untrusted code (which occurs in the similarly modified `ci.yml` job). I'm going to close this PR.

**Comment by CagtayFabry** [comment]
Thank you for the insight @braingram , that is unfortunate to hear

**Pull request closed without merging**
